### ISSUE TITLE
Bootstrap v4.4.1 collapse doesn't work with jQuery v3.5.0

### ISSUE THREAD

**Issue by waclaw66** [twbs/bootstrap#30553]
Hello together,

after upgrade to jQuery 3.5.0 I'm getting an exception 
`TypeError: Cannot convert object to primitive value`
while clicking to navbar-toggler button

It leads to the line
https://github.com/twbs/bootstrap/blob/dca1ab7d877bc4b664b43604657a2b5fbe2b4ecb/js/src/collapse.js#L346

Works fine with jQuery 3.4.1.

Any system and browser.

**Comment by xpressabhi** [comment]
Not working for me as well, by mistake just created new issue.

**Comment by sharkykh** [comment]
Same for Bootstrap 3.4.1.
https://github.com/twbs/bootstrap/blob/68b0d231a13201eb14acd3dc84e51543d16e5f7e/js/collapse.js#L176

**Comment by alliedarmour** [comment]
I have the same problem, in this case my navbar is not working anymore. It surely is the update to 3.5.0 which is causing the problem.

**Comment by XhmikosR** [comment]
I was actually just going to make a new issue to update jQuery to v3.5.0 in v4-dev.

https://blog.jquery.com/2020/04/10/jquery-3-5-0-released/

I'll check now if we have any failing tests.

**Comment by XhmikosR** [comment]
https://github.com/jquery/jquery/issues/4665

So, it seems they will make a patch release to fix this behavior.

@Johann-S do you think we should do anything in our code? I don't know when jQuery v4.0.0 will be out, but any fixes we make will probably be future proof.

**Comment by XhmikosR** [comment]
I was able to reproduce the error after updating to v3.5.0 and adapting our tests in #30559.

**Comment by aldyahsn** [comment]
For me, the console error says
`TypeError: can't convert config to string`

**Comment by XhmikosR** [comment]
After some digging around, it seems this code exists since 2015: https://github.com/twbs/bootstrap/commit/872e56fcf2a7bb1ec936d9e0a5dfed8d31f87590

Let's wait and see what the jQuery developers will say in https://github.com/jquery/jquery/issues/4665

**Comment by XhmikosR** [comment]
Can someone try https://github.com/twbs/bootstrap/pull/30559?

```
https://deploy-preview-30559--twbs-bootstrap.netlify.com/docs/4.4/dist/js/bootstrap.js
https://deploy-preview-30559--twbs-bootstrap.netlify.com/docs/4.4/dist/js/bootstrap.min.js
https://deploy-preview-30559--twbs-bootstrap.netlify.com/docs/4.4/dist/js/bootstrap.bundle.js
https://deploy-preview-30559--twbs-bootstrap.netlify.com/docs/4.4/dist/js/bootstrap.bundle.min.js
```

Our tests seem to pass there. We'll see if/when we can release v4.4.2, but until that happens or until jQuery releases a new patch version people should be careful with the v3.5.0 update.

**Comment by sharkykh** [comment]
@XhmikosR 
I can test that change you made on Bootstrap **v3.4.1**, so I did, and it works.
I hope the plan is to also backport this fix for Bootstrap v3.

I changed [line 176](https://github.com/twbs/bootstrap/blob/v3.4.1/js/collapse.js#L176) from:
```js
if (!data && options.toggle && /show|hide/.test(option)) options.toggle = false 
```
to:
```js
if (!data && options.toggle && typeof option === 'string' && /show|hide/.test(option)) options.toggle = false 
```
and collapse works.
`$(element).collapse('show')` / `$(element).collapse('hide')`works as well.

**Comment by XhmikosR** [comment]
v3 is EOL. I doubt we'll backport this, sorry.

**Comment by alazark94** [comment]
`if (!data && _config.toggle && /show|hide/.test(config))`

should be

`if (!data && _config.toggle && /show|hide/.test(_config))`

**Comment by XhmikosR** [comment]
True I had trouble when I tried that but tests were broken due to other
jQuery changes.

I'll try again later in my branch.

**Comment by XhmikosR** [comment]
Nope, doesn't work because `_config` is still an Object.

**Comment by alazark94** [comment]
I've tried it multiple times and it's working for me.

**Comment by waclaw66** [comment]
`_config` works for me too.
> 
> 
> Nope, doesn't work because `_config` is still an Object.

**Comment by nixprosoft** [comment]
Same error on popover's init with Bootstrap 3.4.1!

**Comment by XhmikosR** [comment]
>I've tried it multiple times and it's working for me.

Yeah, but our test suite doesn't work.

**Comment by XhmikosR** [comment]
Everyone interested in this should follow our #30559 PR and the upstream jQuery issue <https://github.com/jquery/jquery/issues/4665>.

It seems the jQuery developers plan to release a fix for the error that is affecting us, but my plan is to also land #30559 for v4.

**Comment by yickson** [comment]
Downgrade to Jquery 3.4.1 and work again, thanks.

**Comment by kevariable** [comment]
@yickson isn't work with me.
here's my Dependencies

```js
"bootstrap": "^4.4.1",
"jquery": "^3.4.1",
"popper.js": "^1.16.1"
```

**Comment by yickson** [comment]
> @yickson isn't work with me.
> here's my Dependencies
> 
> ```js
> "bootstrap": "^4.4.1",
> "jquery": "^3.4.1",
> "popper.js": "^1.16.1"
> ```

Bootstrap 4.4.1 Not work with Jquery 3.4.1, unfortunately, I use bootstrap 3.3.7 with Jquery 3.4 by compatibility.

**Comment by alazark94** [comment]
Change it to
`“jquery”: “3.4.1”`

Just remove the ‘^’ character

**Comment by kevariable** [comment]
@alazark94 thanks bruh!

**Comment by JiresAng** [comment]
Try to change your v3.5.0 to 3.4.1, downgrade your jQuery

**Comment by JiresAng** [comment]
> @alazark94 thanks bruh!, work as fine

your toggle button works now?

**Comment by valentijnscholten** [comment]
I have the same issue with popover. With jQuery 3.4.1 it works and the `dataAttributes` variable is a full javascript object, which has the `hasOwnProperty` as function. In jQuery 3.5.0 the `dataAttributes` variable is just a dictionary or whatever I should call it. It has no object functions such as `hasOwnProperty`.

![image](https://user-images.githubusercontent.com/4426050/79310571-794a6b00-7efc-11ea-970c-01a5fd34fc54.png)

![image](https://user-images.githubusercontent.com/4426050/79310580-7f404c00-7efc-11ea-8fb0-905419a30201.png)

**Comment by nicholastic** [comment]
jQuery 3.5.0 also won't work with Bootstrap 3.4.1. Dropdowns and collapse don't work. Reverted to jQuery 3.4.1 to make it work.

**Comment by XhmikosR** [comment]
Fixed in #30559.

**Comment by XhmikosR** [comment]
jQuery [v3.5.1](https://blog.jquery.com/2020/05/04/jquery-3-5-1-released-fixing-a-regression/) was released a few hours ago. This should work with Bootstrap 4.x fine.
